cmd/bootkube/render.go

```diff
@@ -111,12 +116,39 @@ func flagsToAssetConfig() (c *asset.Config, err error) {
 			return nil, err
 		}
 	}
+
+	_, podNet, err := net.ParseCIDR(renderOpts.podCIDR)
+	if err != nil {
+		return nil, err
+	}
+
+	_, serviceNet, err := net.ParseCIDR(renderOpts.serviceCIDR)
+	if err != nil {
+		return nil, err
+	}
+
+	if podNet.Contains(serviceNet.IP) || serviceNet.Contains(podNet.IP) {
+		return nil, fmt.Errorf("Pod CIDR %s and service CIDR %s must not overlap", podNet.String(), serviceNet.String())
+	}
+
+	dnsServiceIP, err := pickDNSServiceIP(renderOpts.serviceCIDR)
+	if err != nil {
+		return nil, err
+	}
+
+	if !renderOpts.selfHostKubelet {
+		fmt.Fprintf(os.Stderr, "Host kubelet MUST use --cluster-dns=%s\n", dnsServiceIP)
```

Comment:
This is somewhat unfortunate - because the message isn't very actionable. Can we be more explicit. And I'm going to agree with eric on special casing this if it changes from the default:

Maybe something like:

```
NOTE:
You have selected a non-default service-cidr of `10.10.0.0/24` - be sure that your kubelet service file is updated to use `--cluster-dns=%s`.
```

Once this merges I'll create an issue to follow up where we just generate the kubelet.service file for them - nothing too specific about it anymore (and people can use their own if they want).